### PR TITLE
4th guide: Dynamic filesystem management

### PR DESCRIPTION
**Pull request by tannazie**
guide for modifying template and schema to use a CustomEntity

### CODE REVIEW

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/5169d2f3842d33497671270ac46a66f8280db983/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/15b208fffd0eee268572349a038f5ef36bc68062/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/d23e78b36dad320faf70ea24e207761d3803625d/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/07fe884750f46dc4a02043ee0fdca20a76aa109c/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/f9ac6b3cc8def1a0f497688886cb8b5a8a5a81bf/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/8f2f2063b402648a30e052cf84e6f104e949ce48/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/9b928faea33804f46abb3a6d6bad287c7ac2ebb0/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/92ce148cb6c2f938e0f9e5d19cec9b9ce4ec61bc/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/1f826ca07ab4bc185d485d78f667082160221a81/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/ffc6680218162cfbd260903a9358e8ac3a00e827/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/5aa08fffef9f507596470b291d315b4c4054ed38/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/5f15eddc967dedaf06e4f9ac13c5f9ad537a778c/index.html)

File: docs/images/toolkit/learning-resources/guides/dynamic_filesystem_configuration/placeholder [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (willis102) [inline docs/images/toolkit/learning-resources/guides/dynamic_filesystem_configuration/placeholder]: Do we still need this placeholder file?

File: docs/en/toolkit/learning-resources/guides/dynamic_filesystem_configuration.md [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+
+`<project>/assets/<CustomEntity01>/<asset_type>/<asset>/<step>`
+
+The Set entity is represented as `CustomEntity01`. While we gave CustomEntity01 the *display name* of Set in Shotgun, in our configuration, we’ll always refer to it by its system name, `CustomEntity01`. 
+
+### How the schema uses YAML files
+
+A schema can contain static and dynamic folders. If you have a static folder in your schema called “assets”, it will correspond to a single folder in your production filesystem called “assets”. On the other hand, you might have a dynamic folder in your schema named “asset” that represents one folder for each asset in your project. Every dynamic folder has a corresponding YAML file next to it with the same name as the folder (eg, `asset/` and `asset.yml`) that defines the rules for how to generate production folders based on the schema folder.
+
+ ### Create the new folder and YAML file for the Set entity
```
Comment (willis102) [inline docs/en/toolkit/learning-resources/guides/dynamic_filesystem_configuration.md]: This isn't getting parsed as a heading by kramdown, I think you just need to remove the leading space.

File: docs/images/toolkit/learning-resources/guides/dynamic_filesystem_configuration/placeholder (hunk 1 above)
Comment (willis102) [inline docs/images/toolkit/learning-resources/guides/dynamic_filesystem_configuration/placeholder]: Do we still need this placeholder file?

File: docs/en/toolkit/learning-resources/guides/dynamic_filesystem_configuration.md [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+
+Select **+New File**.
+
+You can create a simple 3D object or just save the file using the **Shotgun > Save File**.
+
+![Python app](../../../../images/toolkit/learning-resources/guides/dynamic_filesystem_configuration/file_save.png)
+
+Success!
+
+Notice the **File Save** dialog box is displaying **Preview: Dining-Room_scene.v001.ma** using the variables that were set in the `preview:`
```
Comment (willis102) [inline docs/en/toolkit/learning-resources/guides/dynamic_filesystem_configuration.md]: Tiny nitpick, maybe the colon should be after the closing backtick?  Sorry if this is too minor to be worth noting!

File: docs/en/toolkit/learning-resources/guides/dynamic_filesystem_configuration.md [hunk 4]
```diff
@@ -0,0 +1,398 @@
...
+
+**Templates** allow you to dynamically name and save files as they’re created using Shotgun data and information from the schema structure. The Default Configuration provides a set of starter templates that you can edit to meet the needs of your pipeline. 
+
+{% include info title="Note" content="The Basic setup for Shotgun integrations doesn’t include filesystem management. In order to configure filesystem management for your project, your project will need and Advanced setup. The first guide, [Getting started with configurations](./advanced_config.md) goes through the Advanced setup process" %}
```
Comment (willis102) [inline docs/en/toolkit/learning-resources/guides/dynamic_filesystem_configuration.md]: `will need and Advanced setup` ->  `will need an Advanced setup`

File: docs/en/toolkit/learning-resources/guides/dynamic_filesystem_configuration.md (hunk 2 above)
Comment (tannazie) [reply docs/en/toolkit/learning-resources/guides/dynamic_filesystem_configuration.md]: good catch. fixed.

File: docs/images/toolkit/learning-resources/guides/dynamic_filesystem_configuration/placeholder (hunk 1 above)
Comment (tannazie) [reply docs/images/toolkit/learning-resources/guides/dynamic_filesystem_configuration/placeholder]: nope. deleted.

File: docs/en/toolkit/learning-resources/guides/dynamic_filesystem_configuration.md (hunk 3 above)
Comment (tannazie) [reply docs/en/toolkit/learning-resources/guides/dynamic_filesystem_configuration.md]: this actually didn't make any sense. fixed up the sentence.

File: docs/en/toolkit/learning-resources/guides/dynamic_filesystem_configuration.md (hunk 4 above)
Comment (tannazie) [reply docs/en/toolkit/learning-resources/guides/dynamic_filesystem_configuration.md]: good catch! fixed.

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/61c810b31bd3226e1465d685e8801f7474a13cfa/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/d09b58d52d88d8674d3c5f349bc2514cc4dc574e/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/d0a6bab05ff69260fd0994668320aab31c2ea476/index.html)

**Pull request merged**